orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/WorkDaoImpl.java

```diff
@@ -368,18 +368,23 @@ public WorkEntity merge(WorkEntity entity) {
     }
 
     @Override
-    public List<Object[]> getWorksByOrcid(String orcid) {
-        //TODO @DanielPalafox fetch only the top_contributors_json after CLI finishes
+    public List<Object[]> getWorksByOrcid(String orcid, boolean topContributorsTogglz) {
+        String contributorsSQL = null;
+        if (topContributorsTogglz) {
+            contributorsSQL = " w.top_contributors_json ";
+        } else {
+            contributorsSQL = " (SELECT to_json(array_agg(row_to_json(t))) FROM (SELECT json_array_elements(json_extract_path(contributors_json, 'contributor')) AS contributor FROM work WHERE work_id=w.work_id limit 100) t) as contributors, ";
```

Comment:
@DanielPalafox the comma at the end of this line will break the query right? could you please remove it?

Thanks

Reply:
Ok, I will remove the unused comma